cf_xarray/accessor.py

```diff
@@ -857,11 +857,20 @@ def _process_signature(
             elif sig.parameters[param].kind is inspect.Parameter.VAR_POSITIONAL:
                 var_args.append(param)
 
-        posargs = []
+        # Catch DataArrays and return them as they are
+        args = list(args)
+        arguments = {}
+        posargs = [
+            args.pop(i) for i, arg in enumerate(args) if isinstance(arg, DataArray)
+        ]
+        for key, value in dict(kwargs).items():
+            if isinstance(value, DataArray):
+                arguments[key] = kwargs.pop(key)
+
         if args or kwargs:
             bound = sig.bind(*args, **kwargs)
-            arguments = self._rewrite_values(
-                bound.arguments, key_mappers, tuple(var_kws)
+            arguments.update(
+                **self._rewrite_values(bound.arguments, key_mappers, tuple(var_kws))
```

Comment:
i'm wondering if it would be cleaner to modify `_rewrite_values` instead.

Reply:
Right! Looks like adding an isinstance check in `_rewrite_values` and `apply_mapper` works well...